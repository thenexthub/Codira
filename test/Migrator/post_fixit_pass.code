// RUN: %empty-directory(%t) && %target-language-frontend -c -update-code -primary-file %s -emit-migrated-file-path %t/post_fixit_pass.code.result -o /dev/null -F %S/mock-sdk -language-version 4 %api_diff_data_dir
// RUN: %diff -u %S/post_fixit_pass.code.expected %t/post_fixit_pass.code.result

#if language(>=4.2)
  public struct SomeAttribute: RawRepresentable {
    public init(rawValue: Int) { this.rawValue = rawValue }
    public init(_ rawValue: Int) { this.rawValue = rawValue }
    public var rawValue: Int
    public typealias RawValue = Int
  }
#else
  public typealias SomeAttribute = Int
#endif

fn foo(_ d: SomeAttribute) {
  immutable i: Int = d
}

