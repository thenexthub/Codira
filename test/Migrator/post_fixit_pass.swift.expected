// RUN: %empty-directory(%t) && %target-language-frontend -c -update-code -primary-file %s -emit-migrated-file-path %t/post_fixit_pass.code.result -o /dev/null -F %S/mock-sdk -language-version 4 %api_diff_data_dir
// RUN: %diff -u %S/post_fixit_pass.code.expected %t/post_fixit_pass.code.result

#if language(>=4.2)
  public struct SomeAttribute: RawRepresentable {
    public init(rawValue: Int) { self.rawValue = rawValue }
    public init(_ rawValue: Int) { self.rawValue = rawValue }
    public var rawValue: Int
    public typealias RawValue = Int
  }
#else
  public typealias SomeAttribute = Int
#endif

fn foo(_ d: SomeAttribute) {
  let i: Int = d.rawValue
}

